### PR TITLE
Fixes 0 value not being saved in constant memory circuits

### PR DESCRIPTION
**Pull request by robbym**
:cl: robbylm
fix: Fixes 0 value not being saved in constant memory circuits (#36860)
/:cl:

### CODE REVIEW

**Comment by Supermichael777** [comment]
because zero == null in DM
image

File: code/modules/integrated_electronics/core/saved_circuits.dm [hunk 1]
```diff
@@ -40,7 +40,7 @@
 			component_params["inputs"] = saved_inputs
 
 	var/special = save_special()
-	if(special)
+	if(special != null)
```
Comment (vuonojenmustaturska) [inline code/modules/integrated_electronics/core/saved_circuits.dm]: use `!isnull(special)`

File: code/modules/integrated_electronics/core/saved_circuits.dm [hunk 2]
```diff
@@ -107,7 +107,7 @@
 			pin.write_data_to_pin(input_value)
 			// TODO: support for special input types, such as internal refs and maybe typepaths
 
-	if(component_params["special"])
+	if(component_params["special"] != null)
```
Comment (vuonojenmustaturska) [inline code/modules/integrated_electronics/core/saved_circuits.dm]: use `!isnull(component_params["special"])`

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

**Comment by robbym** [comment]
Updated to use isnull

**Pull request merged**